client/chain_registry/chain_info.go

```diff
@@ -156,33 +162,36 @@ func (c ChainInfo) GetRandomRPCEndpoint(ctx context.Context) (string, error) {
 	}
 
 	randomGenerator := rand.New(rand.NewSource(time.Now().UnixNano()))
-	return rpcs[randomGenerator.Intn(len(rpcs))], nil
+	endpoint := rpcs[randomGenerator.Intn(len(rpcs))]
+	c.log.Info("Endpoint selected",
+		zap.String("chain_name", c.ChainName),
+		zap.String("endpoint", endpoint),
+	)
+	return endpoint, nil
 }
 
 func (c ChainInfo) GetAssetList(ctx context.Context) (AssetList, error) {
-	cl := github.NewClient(http.DefaultClient)
-
-	chainFileName := path.Join(c.ChainName, "assetlist.json")
-	ch, _, res, err := cl.Repositories.GetContents(
-		ctx,
-		"cosmos",
-		"chain-registry",
-		chainFileName,
-		&github.RepositoryContentGetOptions{})
-	if err != nil || res.StatusCode != 200 {
+	chainRegURL := fmt.Sprintf("https://raw.githubusercontent.com/cosmos/chain-registry/master/%s/assetlist.json", c.ChainName)
+
+	res, err := http.Get(chainRegURL)
+	if res.StatusCode != 200 {
+		return AssetList{}, fmt.Errorf("response code: %d: GET failed: %s", res.StatusCode, chainRegURL)
+	}
+	if err != nil {
```

Comment:
After a call to `http.Get` we need to explicitly close the `res.Body` when we are done with it.
We may get a valid response back from the call to `http.Get` so no error will be returned but we are checking the `StatusCode` before the error which means we could have a resource leak if we don't check the error first and call `defer res.Body.Close()` before checking the `StatusCode`

We could also be comparing the `StatusCode` to one of the constants from the `http` package to increase readability. https://pkg.go.dev/net/http#pkg-constants

ex: 

```
res, err := http.Get(chainRegURL)
if err != nil {
    return AssetList{}, err
}
defer res.Body.Close()

if res.StatusCode != http.StatusOK {
    return AssetList{}, fmt.Errorf("response code: %d: GET failed: %s", res.StatusCode, chainRegURL)
}
```